cranelift/codegen/src/isa/x64/inst.isle

```diff
@@ -748,18 +748,77 @@
 (decl put_in_reg_mem (Value) RegMem)
 (extern constructor put_in_reg_mem put_in_reg_mem)
 
+;; Addressing modes.
+
 (type SyntheticAmode extern (enum))
 
 (decl synthetic_amode_to_reg_mem (SyntheticAmode) RegMem)
 (extern constructor synthetic_amode_to_reg_mem synthetic_amode_to_reg_mem)
 
+(decl amode_to_synthetic_amode (Amode) SyntheticAmode)
+(extern constructor amode_to_synthetic_amode amode_to_synthetic_amode)
+
 (type Amode extern (enum))
 
+(decl amode_imm_reg (u32 Gpr) Amode)
+(extern constructor amode_imm_reg amode_imm_reg)
+
 (decl amode_imm_reg_reg_shift (u32 Gpr Gpr u8) Amode)
 (extern constructor amode_imm_reg_reg_shift amode_imm_reg_reg_shift)
 
-(decl amode_to_synthetic_amode (Amode) SyntheticAmode)
-(extern constructor amode_to_synthetic_amode amode_to_synthetic_amode)
+;; A helper to check if a shift amount (the `Value`) is both constant and
+;; less-than or equal to 3; this is needed since x64 can only shift addresses
+;; using two bits.
+(decl small_const_shift (u8) Value)
```

Comment:
This would be a little clearer with the specific condition in the name I think -- maybe `const_shift_lt_eq_3` or somesuch?

Reply:
Done; it would be nice to have arithmetic expressions in ISLE somehow, a la `(shift <= 3)`.